### ISSUE TITLE
Pre-scripts doesn't persist environment variables

### ISSUE THREAD

**Issue by rasecaccedo** [a85/POSTMan-Chrome-Extension#5170]
Please read through the [guidelines](https://github.com/postmanlabs/postman-app-support#guidelines-for-reporting-issues) before creating a new issue.

**Describe the bug**
The environment variables setted in pre-scripts do not persist for further requests. Sample code:
Request 1:
Pre-request Script
```
var timestamp1 = Date.now();
pm.environment.set("timestampHeader", timestamp1);
console.log(pm.environment.get("timestampHeader")); // 1536849330726
```
Body
```
{
  "uid":"{{userId}}",
  "timestamp":"{{timestampHeader}}"
}
```
Test
```
pm.environment.set("test", Data.now());
console.log(pm.environment.get("test")); // 1536849330756
console.log(pm.environment.get("timestampHeader")); // 1536849330726
```
Request 2:
Pre-request Script
```
console.log(pm.environment.get("test")); //1536849330756
console.log(pm.environment.get("timestampHeader")); //undefined
```

**To Reproduce**
Steps to reproduce the behaviour:
1. Create a request
2. Add a pre-request script in which an env variable is setted
3. Add a test in which an env variable is setted
4. Create a second request
5. Try to use the env variable setted in the previous request pre-request script

**Expected behaviour**
The environment variable has the proper value defined during the first request pre-request execution.

**App information (please complete the following information):**
 - App Type Mac Native App
 - Postman Version: 6.2.5
 - OS: macOS Sierra 10.12.6.

**Comment by numaanashraf** [comment]
@rasecaccedo Could you verify whether "Automatically persist variable values" is turned on in Settings > General?

**Comment by rasecaccedo** [comment]
> @rasecaccedo Could you verify whether "Automatically persist variable values" is turned on in Settings > General?

@numaanashraf yes it is turned on, i'm attaching an screenshot 
<img width="342" alt="screen shot 2018-09-17 at 09 08 44" src="https://user-images.githubusercontent.com/1792105/45609719-6722f500-ba59-11e8-9918-dd4c0908035a.png">

**Comment by numaanashraf** [comment]
@rasecaccedo We are having trouble replicating this at our end. Could you share a sample collection?

**Comment by rasecaccedo** [comment]
@numaanashraf i did test it with version 6.4.4 and it seems to be working.
I will review just in case i didnt test it well and get back if i found something.

Thanks for the support!

**Comment by numaanashraf** [comment]
@rasecaccedo Thanks for letting us know. Feel free to re-open this issue if you are still facing it.

**Comment by ahsankarimbhai** [comment]
@numaanashraf I am seeing the exact same issue with latest Postman 7.0.7 on my MacOS 10.13.2. I tried workaround as mentioned here https://github.com/postmanlabs/postman-app-support/issues/5039#issuecomment-420258338 

I did a fresh install and it still doesn't work. Can you suggest some other workaround?

**Comment by Jamiewarb** [comment]
I also seem to be encountering this issue on 7.2.0

**Comment by ttibensky** [comment]
still having this issue in Version 7.3.4

**Comment by abbas-v1** [comment]
I don't know if you have the same issue but mine had a carriage return at the end of the variable name.

**Comment by coupa-anya** [comment]
I had this issue, and in the process of testing why some things worked and others didn't, I found that if I cleared the cookies for my service in my script, any variables set after that point would appear to be set within the script, but would not persist after the script exits.

Repro:

```
pm.cookies.jar().clear(pm.environment.get('my_service_uri'))
pm.environment.set('some_variable', 'a')
console.log(pm.environment.get('some_variable'))
```

Result: The log line prints 'a' as expected, but 'some_variable' is not set in the environment after the request is executed.

However, if I simply set the variable before clearing the cookies:
```
pm.environment.set('some_variable', 'a')
pm.cookies.jar().clear(pm.environment.get('my_service_uri'))
console.log(pm.environment.get('some_variable'))
```

Result:  The log line prints 'a' as expected, AND 'some_variable' is set correctly after the request is executed.

At least for my case, it appears that something in the cookie clearing code interferes with the correct persistence of variables.

Postman Version 7.26.1 on MacOS Catalina
